src/main/java/net/openhft/chronicle/core/Jvm.java

```diff
@@ -406,7 +406,14 @@ public long usedDirectMemory() {
 
     private static long maxDirectMemory0() {
         try {
-            Class<?> clz = Class.forName("sun.misc.VM");
+            Class<?> clz;
+
+            if (IS_JAVA_9_PLUS) {
```

Comment:
If you're concerned about ordering, just call `getMajorVersion0()` and check the result. This method should only be invoked once, so it's not like there's going to be a large cost associated with it.

Reply:
@epickrram There is another solution: Just move initialization of of static fields to `static {}` block.

The original code:

```
    private static final int JVM_JAVA_MAJOR_VERSION = getMajorVersion0();
    private static final boolean IS_JAVA_9_PLUS = JVM_JAVA_MAJOR_VERSION > 8;
    private static final long MAX_DIRECT_MEMORY = maxDirectMemory0();

    static {

```

Proposed code:

```
    private static final int JVM_JAVA_MAJOR_VERSION;
    private static final boolean IS_JAVA_9_PLUS;
    private static final long MAX_DIRECT_MEMORY;

    static {
        JVM_JAVA_MAJOR_VERSION = getMajorVersion0();
        IS_JAVA_9_PLUS = JVM_JAVA_MAJOR_VERSION > 8;
        MAX_DIRECT_MEMORY = maxDirectMemory0();
```

This way the risk of introducing a bug is lower.

Reply:
That would work. Please include a comment stating that the result of `IS_JAVA_9_PLUS` is used by `maxDirectMemory0`.

Reply:
@epickrram Done